formidable/forms/__init__.py

```diff
@@ -48,10 +48,44 @@ def __init__(self, *args, **kwargs):
         super(BaseDynamicForm, self).__init__(*args, **kwargs)
         self._bound_fields_cache = FormidableBoundFieldCache()
 
+    def get_removed_fields(self, cleaned_data):
+        """
+        Build the list of fields to be removed due to conditional displays
+        """
+        # build a catalog of fields **targeted** by the conditions
+        condition_targets = {}
+
+        # For each condition, extract its status (should I display or not)
+        for condition in self._conditions:
+            # should we keep these fields?
+            keep_fields = condition.keep_fields(cleaned_data)
+            for field_id in condition.fields_ids:
+                # Fill the catalog
+                if field_id not in condition_targets:
+                    condition_targets[field_id] = []
+                condition_targets[field_id].append(keep_fields)
+        # Here, the catalog contains fields targeted by 1 or many conditions.
+
+        # If only one condition says "please display X", we'll keep X
+        # That's why we gather the conditions using "any"
+        condition_targets = {k: any(v) for k, v in condition_targets.items()}
+        # We'll only remove fields that are targeted by conditions **and**
+        # those conditions are false
+        return (k for k, v in condition_targets.items() if not v)
```

Comment:
So you don't want to remove intermediate variable? And write just a 
`(k for k, v in condition_targets.items() if not any(v)`

Reply:
IMHO the intermediate variable is the way to make sure that you can understand what it does. It's very easy to overlook a comprehension list and misread. ``if not v`` and ``if not any(v)`` are very close. Code is more often read than written.
If you have two filters/actions to apply to an iterable, I prefer having two lines of code, each one doing one thing and doing it clearly.
And one can inspect the intermediate variable at debugging time.

Reply:
Ok, I see. Maybe it's just because I'm in love with this python comprehension list/dict feature and I don't like to add extra lines in this case :)

Reply:
:)